### PR TITLE
Move quoted null and unit encoding to internal

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/internal/quoted/Expr.scala [hunk 1]
```diff
@@ -55,4 +55,16 @@ object Expr {
...
+  def `null`: QuoteContext ?=> quoted.Expr[Null] = qctx ?=> {
+    import qctx.tasty._
+    Literal(Constant(null)).seal.asInstanceOf[quoted.Expr[Null]]
+  }
+
+  /** Returns a unit expresssion equivalent to `'{}` or `'{()}` */
+  def Unit: QuoteContext ?=> quoted.Expr[Unit] = qctx ?=> {
+    import qctx.tasty._
+    Literal(Constant(())).seal.asInstanceOf[quoted.Expr[Unit]]
+  }
```
Comment (liufengyun) [inline library/src/scala/internal/quoted/Expr.scala:68]: It seems both can be `val` instead of `def`?

Comment (nicolasstucki) [reply library/src/scala/internal/quoted/Expr.scala:68]: They could but they would not get the optimization where the context parameter is passed directly as one of the method parameters and the closure is not created. We effectively call the following method at runtime (not closure creation needed):
```scala
def Unit(using q: QuoteContext): quoted.Expr[Unit] = {
  import qctx.tasty._
  Literal(Constant(())).seal.asInstanceOf[quoted.Expr[Unit]]
}
```

Comment (liufengyun) [reply library/src/scala/internal/quoted/Expr.scala:68]: Good point, maybe we can write it directly, instead of the original form?

Comment (nicolasstucki) [reply library/src/scala/internal/quoted/Expr.scala:68]: It is like this to align with the quote internal encoding to simplify the transformation.

Comment (nicolasstucki) [reply library/src/scala/internal/quoted/Expr.scala:68]: I will consider changing the encoding but this is a larger change that will take some time to accomplish.

Comment (liufengyun) [reply library/src/scala/internal/quoted/Expr.scala:68]: Thanks for the clarification, it makes sense.

**Pull request merged**